src/commands/service/create-revision.ts

```diff
@@ -91,55 +58,133 @@ export const commandDecorator = (command: commander.Command): void => {
-      // Make a PR against all default rings
-      for (const ring of defaultRings) {
-        if (typeof title !== "string") {
-          title = `[SPK] ${sourceBranch} => ${ring}`;
-          logger.info(`--title not set, defaulting to: '${title}'`);
-        }
-        await createPullRequest(title, sourceBranch, ring, {
-          description,
-          orgName,
-          originPushUrl: remoteUrl,
-          personalAccessToken
-        });
-      }
       await exitCmd(logger, process.exit, 0);
     } catch (err) {
       logger.error(err);
       await exitCmd(logger, process.exit, 1);
     }
   });
 };
+
+/**
+ * Gets the default rings
+ * @param targetBranch Target branch/ring to create a PR against
+ * @param bedrockConfig The bedrock configuration file
+ */
+export const getDefaultRings = (
+  targetBranch: string | undefined,
+  bedrockConfig: IBedrockFile
+): string[] => {
+  const defaultRings: string[] = targetBranch
+    ? [targetBranch]
+    : Object.entries(bedrockConfig.rings || {})
+        .map(([branch, config]) => ({ branch, ...config }))
+        .filter(ring => !!ring.isDefault)
+        .map(ring => ring.branch);
+  if (defaultRings.length === 0) {
+    throw Error(
+      `Default branches/rings must either be specified in ${join(
+        __dirname,
+        "bedrock.yaml"
+      )} or provided via --target-branch`
+    );
+  }
+  logger.info(
+    `Creating pull request against branches: ${defaultRings.join(", ")}`
+  );
+  return defaultRings;
+};
+
+/**
+ * Gets the source branch or parses git for the source branch
+ * @param sourceBranch The source branch
+ */
+export const getSourceBranch = async (
+  sourceBranch: string | undefined
+): Promise<string | undefined> => {
+  if (
+    typeof sourceBranch !== "string" ||
+    (typeof sourceBranch === "string" && sourceBranch.length === 0)
+  ) {
+    // Parse the source branch from options
+    // If it does not exist, parse from the git client
+    logger.info(
+      `No source-branch provided, parsing the current branch for git client`
+    );
+    sourceBranch = await getCurrentBranch();
+    if (sourceBranch.length === 0) {
+      throw Error(
+        `Zero length branch string parsed from git client; cannot automate PR`
+      );
+    }
+  }
+  return sourceBranch;
+};
+
+/**
+ * Creates a pull request from the given source branch
+ * @param defaultRings List of default rings
+ * @param title Title of pr
+ * @param sourceBranch Source branch for pr
+ * @param description Description for pr
+ * @param orgName Organization name
+ * @param remoteUrl Remote url
+ * @param personalAccessToken Access token
+ */
+export const makePullRequest = async (
+  defaultRings: string[],
+  title: string | undefined,
+  sourceBranch: string | undefined,
+  description: string | undefined,
+  orgName: string | undefined,
+  remoteUrl: string | undefined,
+  personalAccessToken: string | undefined
+) => {
```

Comment:
Optionally, you can use this function to help you to check for valid option values. https://github.com/CatalystCode/spk/blob/master/src/lib/commandBuilder.ts#L67

Reply:
There are no required values though.

Reply:
e.g. `remoteUrl`, `sourceBranch`, etc are required values, either they are provided by user in commandline or we get them from config.yaml.

Reply:
minor thing, we can do this later. thanks